### PR TITLE
Adding Site Daily Metrics code

### PR DESCRIPTION
**Pull request by johnbaldwin**
Added SiteDailyMetrics model, serializer, filter, view and tests

"Cleanliness" in testing issue: retrieving 'values' for a query set retain the `date time.datetime` and `date time.date` objects in the resulting values, therefore I converted the serialized dates to a datetime or date object to compare in the tests (see `test_site_daily_metrics_view.py` and `test_serializers.py`

I know there is a clean solution in Django 2.0 and documented that in `test_serializers.py`

Just to note, I'm not going for exhaustive testing at this point, really just to get a well rounded base for the different parts of the code to make additional test writing much faster

And hey! We've now got only 15 files in this review. Down 5 from last review!

Also note that this branch is to merge into `john/user-index`, which is to merge into `develop`. This is because I'm chaining the branches to avoid extra work in duplicating support code

### CODE REVIEW

File: edx_figures/models.py [hunk 1]
```diff
@@ -4,3 +4,31 @@
...
+
+    # Date for which this record's data are collected
+    date_for = models.DateField(unique=True)
+    cumulative_active_user_count = models.IntegerField(blank=True, null=True)
+    todays_active_user_count = models.IntegerField(blank=True, null=True)
+    total_user_count = models.IntegerField()
+    course_count = models.IntegerField()
+    total_enrollment_count = models.IntegerField()
+
+    # Foreign key relationships
```
Comment (johnbaldwin) [inline edx_figures/models.py]: Intentionally left commented as a placeholder to remind me to add Site

File: edx_figures/views.py [hunk 2]
```diff
@@ -99,3 +105,23 @@ def get_queryset(self):
         queryset = super(UserIndexView, self).get_queryset()
 
         return queryset
+
+##
+## Views for edX Figures models
+##
+
+#class SiteDailyMetricsViewSet(CommonAuthMixin, viewsets.ModelViewSet):
```
Comment (johnbaldwin) [inline edx_figures/views.py]: Remarked out when copied over from my sandbox dev. I will add Auth toward the end of this sprint after I get more of the metrics code in here

File: edx_figures/admin.py [hunk 3]
```diff
@@ -1,3 +1,15 @@
 from django.contrib import admin
 
 # Register your models here.
+
+from .models import (
+    SiteDailyMetrics,
+
```
Comment (johnbaldwin) [inline edx_figures/admin.py]: To note, this blank line will be filled with `CourseDailyMetrics` shortly

File: edx_figures/migrations/0001_initial.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
```
Comment (johnbaldwin) [inline edx_figures/migrations/0001_initial.py]: @OmarIthawi Just remembered. I'll rename this file to be more descriptive

File: tests/test_models.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    @pytest.fixture(autouse=True)
+    def setup(self, db):
+        '''
+
+        '''
+        self.site_daily_metrics = [
+            SiteDailyMetricsFactory()
+        ]
+
+    def test_foo(self):
```
Comment (johnbaldwin) [inline tests/test_models.py]: I'll remove this test

**Comment by bryanlandia** [comment]
@johnbaldwin There are conflicts in the merge (`devsite/requirements.txt`).  Please resolve and push again.

File: edx_figures/migrations/0001_initial.py (hunk 4 above)
Comment (OmarIthawi) [reply edx_figures/migrations/0001_initial.py]: @johnbaldwin initial is good enough. Subsequent migrations are the tricky ones. Thanks though :)

**Comment by johnbaldwin** [comment]
@bryanlandia I addressed the conflicts

**Pull request merged**